WebpageRevamped/src/components/home/Home.tsx

```diff
@@ -34,19 +38,66 @@ const Home = () => {
       fiveThousandViewDataLabels,
       fiveThousandTextDataLabels,
       fiveThousandImageDataLabels,
-    })
-    setShowGraph(true)
+    });
+    setShowGraph(true);
+    setShowSelection(false); // Hide the selection container after generating the report
+  };
+
+  const [width, setWidth] = useState<number>(window.innerWidth);
+
+  function handleWindowSizeChange() {
+    setWidth(window.innerWidth);
   }
 
+  useEffect(() => {
+    window.addEventListener('resize', handleWindowSizeChange);
+    return () => {
+      window.removeEventListener('resize', handleWindowSizeChange);
+    };
+  }, []);
+
+  const isMobile = width <= 768;
+
+  const toggleSelection = () => {
+    setShowSelection(!showSelection); // Toggle the visibility of the selection container
+  };
+
+  const handleClose = () => {
+    setShowSelection(false); // Close the selection container
+  };
+
   return (
-    <div className={'MainContainer'}>
-      <Header />
-      <div className={'HomeContainer'}>
-        <SelectionContainer onGenerateReport={handleGenerateReport} />
-        {showGraph ? <GraphContainer {...graphData} /> : null}
+      <div className={'MainContainer'}>
+        <Header/>
+        <div className={'HomeContainer'}>
+          {/* Icon for mobile view */}
+          {isMobile &&
+              <div className="SelectionToggleContainer">
+                <IconButton onClick={toggleSelection}>
+                  <img src={sideNav} alt={'Dream11 Logo'} width={20} height={20}/>
+                </IconButton>
+              </div>
+          }
+
+          {/* Selection Container */}
+          {isMobile ? (
+              <div
+                  className={`selectionContainerOverlay ${showSelection ? 'show' : ''}`}
+                  onClick={handleClose}
+              >
+                <div className="selectionContainerContent" onClick={(e) => e.stopPropagation()}>
+                  <SelectionContainer onGenerateReport={handleGenerateReport} hideSelection={() => setShowSelection(false)}/>
+                </div>
+              </div>
```

Comment:
can we move this to a different component

Reply:
Fixed